scripts/kconfig/merge_config.py

```diff
@@ -3,50 +3,92 @@
+                else:
+                    config_name = match.group(1)
+                cfg_list.append((config_name, match.group(0), ))
+    return cfg_list
+
+
+def remove_config(initfile_content, cfg):
+    pattern = re.compile("{}[ =]".format(cfg))
+    lines = initfile_content.split("\n")
+    new_lines = []
+
+    for line in lines:
+        match = pattern.match(line)
+        if not match:
+            new_lines.append(line)
+    initfile_content = "\n".join(new_lines)
+    return initfile_content
+
+
+def append_files(f1, initfile_content):
+    fin = open(f1, "r")
+    data2 = fin.read()
+    fin.close()
+    return(initfile_content + data2)
+
+
 def main():
     args = arguments_parse()
+    init_file = open(args.config[0])
+    initfile_content = init_file.read()
+    init_file.close()
+    print("-- Using %s as base" % args.config[0])
 
-    pattern = re.compile('^(CONFIG_[a-zA-Z0-9_]*)[= ].*')
-    config_values = {}
+    for config_file in args.config[1:]:
+        print("-- Merging %s" % config_file)
+        cfg_list = get_config_list(config_file)
 
-    for config_file in args.config[:]:
-        print("Merging %s" % config_file)
-        with open(config_file, 'r') as file:
-            for line in file:
-                match = pattern.match(line)
-                if match:
-                    config_name = match.group(1)
-                    config_full = match.group(0)
-                    if config_name in config_values:
-                        if config_values[config_name] != config_full and not args.quiet:
-                            print("Value of %s is redefined by fragment %s" % (config_name, config_file))
-                            print("Previous value %s" % config_values[config_name])
-                            print("new value %s" % config_full)
-                        elif args.warnredun:
-                            print("Value of %s is redundant by fragment %s" % (config_name, config_file))
-                    config_values[config_name] = config_full
+        for cfg, full in cfg_list:
+            prev=list(filter(lambda x: cfg in x, initfile_content.split("\n")))
+            if len(prev) > 0 and prev[0] != full and not args.quiet:
+                print("-- Value of {} is redefined by fragment {}".format(cfg, config_file))
+                print("-- Previous value: {}".format(prev[0]))
+                print("-- New value: {}".format(full))
+            elif args.warnredun:
+                print("Value of %s is redundant by fragment %s" % (cfg, config_file))
+
+            initfile_content = remove_config(initfile_content, cfg)
+
+        initfile_content = append_files(config_file, initfile_content)
 
     if args.runmake:
         print("Running make not yet supported")
+    else:
+        with open('%s/.config' % args.output_dir, 'w') as file:
+            file.write(initfile_content)
+        with open('%s/.config.test' % args.output_dir, 'w') as file:
```

Comment:
Is this debug code that was included by mistake?

Reply:
yep